src/app/@ansyn/menu-items/tools/components/image-processing-control/image-processing-control.component.ts

```diff
@@ -22,100 +16,49 @@ export interface IImageProcParamComp extends IImageProcParam {
 
+	public manualImageProcessingParams$: Observable<Object> = this.store$.select(toolsStateSelector)
+		.map((tools: IToolsState) => tools.manualImageProcessingParams)
+		.distinctUntilChanged()
+		.filter(Boolean)
+		.do((imageManualProcessArgs) => this.imageManualProcessArgs = imageManualProcessArgs);
 
-	public mapsImapgeProcessingState =	this.store$.select(toolsStateSelector)
-		.withLatestFrom(<Observable<IMapState>>this.store$.select(mapStateSelector))
-		.do((res: [IToolsState , IMapState])   =>
-		{
-			const [toolsState, caseMapState]: [IToolsState , IMapState] = res;
-			if (toolsState.imageProcessingHash[caseMapState.activeMapId] !== undefined) {
-				this.params.forEach(param => {
-					this.params[this.params.findIndex((x) => x.name === param.name)].value = toolsState.imageProcessingHash[caseMapState.activeMapId][param.name];
-				});
-			}
-		});
-
-
-	params: Array<IImageProcParamComp> = this.config.ImageProcParams.map(param => {
-		return { ...param, value: param.defaultValue };
-	});
-
-	// public throttledManualImageProcess: Function;	// throttled function
-
-	@HostBinding('class.expand') @Input()
-	set expand(value) {
-		this._isExpended = value;
+	get params(): Array<IImageProcParam> {
+		return this.config.ImageProcParams;
 	}
 
-	get expand() {
-		return this._isExpended;
+	get defaultImageManualProcessArgs(): ImageManualProcessArgs {
+		return this.params.reduce<ImageManualProcessArgs>((initialObject: any, imageProcParam) => {
+			return <any> { ...initialObject, [imageProcParam.name]: imageProcParam.defaultValue };
+		}, {});
 	}
 
-	@Input()
-	set initParams(_initParams) {
-		if (_initParams) {
-			const isChangeFromInit = this.params.some(({ value, name }) => value !== _initParams[name]);
-			if (isChangeFromInit) {
-				this.params.forEach((param) => param.value = _initParams[param.name]);
-				this.manualImageProcess();
-			}
-		}
-		else {
-			this.resetParams();
-			this.isActive.emit(false);
-		}
-	}
+	imageManualProcessArgs: ImageManualProcessArgs = this.defaultImageManualProcessArgs;
 
-	@Output() isActive = new EventEmitter<boolean>();
+	@HostBinding('class.expand') @Input() expand;
 
 	constructor(public store$: Store<IToolsState>, @Inject(toolsConfig) protected config: IToolsConfig) {
-		// this.throttledManualImageProcess = throttle(this.manualImageProcess, 200);
 	}
 
-	manualImageProcess() {
-
-		const isChangeFromDefualt = this.params.some(({ value, defaultValue }) => value !== defaultValue);
-		let dispatchValue = <ImageManualProcessArgs> {};
-		if (isChangeFromDefualt) {
-			this.params.forEach(param => {
-				dispatchValue[param.name] = param.value;
-			});
-		}
-		else {
-			dispatchValue = undefined;
-		}
-		this.isActive.emit(isChangeFromDefualt);
-		this.store$.dispatch(new SetManualImageProcessing({ processingParams: dispatchValue }));
+	resetOne(paramToReset) {
+		this.updateParam(this.params.find((param) => param.name === paramToReset.name).defaultValue, paramToReset.name);
```

Comment:
defaultImageManualProcessArgs[paramToReset.name] instead of nasty find

Reply:
done